apps/finance/app/src/components/Transfers.js

```diff
@@ -55,15 +56,58 @@ const getFilteredTransfers = ({
           start: startOfDay(selectedDateRange.start),
           end: endOfDay(selectedDateRange.end),
         })) &&
-      (selectedToken === 0 ||
-        addressesEqual(token, tokens[selectedToken - 1].address)) &&
+      (selectedToken === null ||
+        addressesEqual(token, selectedToken.address)) &&
       (transferType === TransferTypes.All ||
         (transferType === TransferTypes.Incoming && isIncoming) ||
         (transferType === TransferTypes.Outgoing && !isIncoming))
   )
 }
+const getDownloadData = async (transfers, tokenDetails, resolveAddress) => {
+  const mappedData = await Promise.all(
+    transfers.map(
+      async ({
+        date,
+        numData: { amount },
+        reference,
+        isIncoming,
+        entity,
+        token,
+      }) => {
+        const { symbol, decimals } = tokenDetails[toChecksumAddress(token)]
+        const formattedAmount = formatTokenAmount(
+          amount,
+          isIncoming,
+          decimals,
+          true,
+          { rounding: 5 }
+        )
+        const { name = '' } = (await resolveAddress(entity)) || {}
+        return `${formatDate(
+          date
+        )},${name},${entity},${reference},${`${formattedAmount} ${symbol}`}`
+      }
+    )
+  )
+  const csvContent = [
+    'data:text/csv;charset=utf-8,Date,Name,Source/Recipient,Reference,Amount',
+  ]
+    .concat(mappedData)
+    .join('\n')
+  return window.encodeURI(csvContent)
+}
+const getDownloadFilename = (dao, { start, end }) => {
+  const today = format(Date.now(), 'yyyy-MM-dd')
+  let filename = `finance_${dao}_${today}.csv`
+  if (start && end) {
+    const formattedStart = format(selectedDateRange.start, 'yyyy-MM-dd')
+    const formattedEnd = format(selectedDateRange.end, 'yyyy-MM-dd')
+    filename = `finance_${dao}_${formattedStart}_to_${formattedEnd}.csv`
+  }
```

Comment:
Nitpick: since we're changing this now, should we handle the case where you have a start date but no end date (so end defaults to today)?